include/micm/process/process_set.hpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2023 National Center for Atmospheric Research,
+//
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include <micm/process/process.hpp>
+#include <micm/solver/state.hpp>
+#include <micm/util/matrix.hpp>
+#include <vector>
+
+namespace micm
+{
+
+  /// @brief Solver function calculators for a collection of processes
+  class ProcessSet
+  {
+    std::vector<std::size_t> number_of_reactants_;
+    std::vector<std::size_t> reactant_ids_;
+    std::vector<std::size_t> number_of_products_;
+    std::vector<std::size_t> product_ids_;
+    std::vector<double> yields_;
+
+   public:
+    /// @brief Default constructor
+    ProcessSet() = default;
+
+    /// @brief Create a process set calculator for a given set of processes
+    /// @param processes Processes to create calcualtor for
+    /// @param state Solver state
+    ProcessSet(const std::vector<Process>& processes, const State& state);
+
+    /// @brief Add forcing terms for the set of processes for the current conditions
+    /// @param rate_constants Current values for the process rate constants (grid cell, process)
+    /// @param state_variables Current state variable values (grid cell, state variable)
+    /// @param forcing Forcing terms for each state variable (grid cell, state variable)
+    void AddForcingTerms(
+        const Matrix<double>& rate_constants,
+        const Matrix<double>& state_variables,
+        Matrix<double>& forcing) const;
+  };
+
+  inline ProcessSet::ProcessSet(const std::vector<Process>& processes, const State& state)
+      : number_of_reactants_(
+            [&]() -> std::vector<std::size_t>
```

Comment:
Since the data member are not const, it might be more readable to move these into the body of the constructor. We would no longer need the lambda functions.